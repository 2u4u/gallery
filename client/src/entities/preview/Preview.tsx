import { PreviewType } from 'shared/types/index.type';
import PreviewDelete from './PreviewDelete';
import PreviewStatus from './PreviewStatus';
import PreviewSubmit from './PreviewSubmit';
import PreviewImage from './PreviewImage';
import PreviewName from './PreviewName';
import './index.scss';

type Props = {
    previews: PreviewType[];
    handleDelete: (file: File) => void;
    handleUpload: () => void;
}

function Preview({ previews, handleDelete, handleUpload }: Props) {
    return (
        <>
            <div className='preview'>
                {previews.map((preview, index) => 
                    <div key={index} className='preview__item'>
                        <div className='preview__block'>
                            <PreviewImage preview={preview} />
                            <PreviewName preview={preview} />
                            <PreviewStatus preview={preview} />
                        </div>
                        <PreviewDelete preview={preview} handleDelete={handleDelete} />
                    </div>)
                }
            </div>

            <PreviewSubmit handleUpload={handleUpload} />
        </>
    );
}

export default Preview;
