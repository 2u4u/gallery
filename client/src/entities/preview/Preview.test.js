import renderer from 'react-test-renderer';
import Preview from './Preview';

describe('Preview component ', () => {
    const previews = [
        {
            type: 'image',
            url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
            name: 'test image',
            file: 'Fake file',
        }
    ];
    const handleDelete = () => false;
    const handleUpload = () => false;
    it('matches snapshot', () => {
        const component = renderer.create(
            <Preview handleDelete={handleDelete} handleUpload={handleUpload} previews={previews} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
