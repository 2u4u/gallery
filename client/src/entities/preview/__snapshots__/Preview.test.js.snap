// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component  matches snapshot 1`] = `
Array [
  <div
    className="preview"
  >
    <div
      className="preview__item"
    >
      <div
        className="preview__block"
      >
        <div
          className="preview__image"
          style={
            Object {
              "backgroundImage": "url(https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png)",
            }
          }
        />
        <div
          className="preview__name"
        >
          test image
        </div>
        <div
          className="preview__status"
        />
      </div>
      <button
        onClick={[Function]}
      >
        <img
          alt="close modal"
          className="preview__close"
          src="./img/close.svg"
        />
      </button>
    </div>
  </div>,
  <div
    className="preview__block preview__block--footer"
  >
    <button
      className="preview__button"
      onClick={[Function]}
      type="button"
    >
      Upload
    </button>
  </div>,
]
`;
